geode-core/src/main/java/org/apache/geode/internal/datasource/GemFireTransactionDataSource.java

```diff
@@ -219,7 +221,13 @@ void registerTranxConnection(XAConnection xaConn) throws Exception {
    * @return ???
    */
   protected Connection getSQLConnection(PooledConnection poolC) throws SQLException {
-    Connection conn = poolC.getConnection();
+    Connection conn;
+    try {
+      conn = poolC.getConnection();
+    } catch (SQLException e) {
+      provider.returnAndExpireConnection(poolC);
+      throw new SQLException(e);
```

Comment:
We typically try to avoid wrapping multiple layers of the same Exception. Is there a reason to wrap the SQLException inside a new instance of SQLException? If not then just rethrow the same e:
```
      provider.returnAndExpireConnection(poolC);
      throw e;
```

Reply:
Updated